### PR TITLE
C# code generation

### PR DESCRIPTION
**Pull request by mmoskal**
This implements `appTarget.nativeType == "C#"`
See https://github.com/Microsoft/pxt-samplecs/blob/master/libs/core/Core.cs for accompanying runtime implementation, which should be integrated in the final target.

It currently passes `lang-test0`. 

While we're not using this yet, I would like it merged, to avoid bitrot (especially in `cpp.ts`, which now also handles parsing C#).

### CODE REVIEW

File: pxtcompiler/emitter/hexfile.ts [hunk 1]
```diff
@@ -155,6 +155,12 @@ namespace ts.pxtc {
 
             patchSegmentHex(hex)
 
+            if (opts.nativeType == "C#") {
```
Comment (pelikhan) [inline pxtcompiler/emitter/hexfile.ts]: Consider having a constant for C#

**Comment by pelikhan** [comment]
@mmoskal did you do a sanity check for other targets?

**Comment by mmoskal** [comment]
Will do tomorrow

________________________________

**Comment by mmoskal** [comment]
Just checked Adafruit - the exact same binary is generated and no changes in shims.d.ts.

**Pull request merged**